internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -163,6 +164,44 @@ func (r *NonAdminBackupStorageLocationReconciler) initNaBSLDelete(ctx context.Co
 	return false, nil
 }
 
+// deleteNonAdminBackups deletes all NonAdminBackups associated with the given NonAdminBackupStorageLocation
+func (r *NonAdminBackupStorageLocationReconciler) deleteNonAdminBackups(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	nonAdminBackupList := &nacv1alpha1.NonAdminBackupList{}
+	listOpts := &client.ListOptions{Namespace: nabsl.Namespace}
+
+	if err := r.Client.List(ctx, nonAdminBackupList, listOpts); err != nil {
+		return false, err
+	}
+
+	if len(nonAdminBackupList.Items) == 0 {
+		logger.V(1).Info("No NonAdminBackups found in NonAdminBackupStorageLocation namespace", "nabsl", nabsl.Name)
+		return false, nil
+	}
+
+	for _, nonAdminBackup := range nonAdminBackupList.Items {
+		// Ensure it belongs to this StorageLocation
+		if nonAdminBackup.Spec.BackupSpec == nil || nonAdminBackup.Spec.BackupSpec.StorageLocation != nabsl.Name {
+			continue
+		}
+
+		logger.V(1).Info("Deleting NonAdminBackup", "backup", nonAdminBackup.Name)
+
+		if err := r.Delete(ctx, &nonAdminBackup); err != nil {
+			if apierrors.IsNotFound(err) {
+				// Ignore NotFound errors (already deleted)
+				continue
+			}
+			logger.Error(err, "Failed to delete NonAdminBackup", "backup", nonAdminBackup.Name)
+			// Continue deleting other backups, this should not block the deletion of the NonAdminBackupStorageLocation
```

Comment:
this comment does not make much sense 

GC will remove orphaned Velero objects in OADP namespace (those non admin can not interact with)

the idea of this PR is to avoid mismatches in NonAdminBackup and Velero Backup

problem: non admin user creates NABSL -> creates NAB -> deletes NABSL -> recreates NABSL. Since BSL created by NABSL will always have a different name, if we do not delete NABSL's NABs when deleting it, non admin user can not use NAB `spec.delete`, because DeleteBackupRequest will reference a BSL that does not exist. To avoid scenarios like this, we are deleting the NABs

So, I think `continue` is not right here

Reply:
So should NABSL controller error on such condition and not remove NABSL finalizer?

Reply:
I think it's ok to continue deleting NABackups here, as we only call delete on those objects. If somethng goes wrong in the deletion then NAB controller will cover that?

Reply:
Changed the logic to not continue.